platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformData.java

```diff
@@ -98,18 +102,22 @@ private static final class ClassVersion {
     private Hash epochHash;
 
     /**
-     * The next epoch hash, used to update the epoch hash at the next round boundary. This field is not part of the
-     * hash and is not serialized.
+     * The next epoch hash, used to update the epoch hash at the next round boundary. This field is not part of the hash
+     * and is not serialized.
      */
     private Hash nextEpochHash;
 
+    /**
+     * The minimum generation of non-ancient events for this state.
+     */
+    private long minimumGenerationNonAncient;
```

Comment:
why duplicate this information if its already there?

Reply:
You can't actually compute this unless you know the setting for the rounds non-ancient. If we recorded the value of the setting for rounds non-ancient here we could recompute it... but I thought the direct value `minimumGenerationNonAncient` would be preferable.

Reply:
maybe better to write the setting

Reply:
I now write this setting instead of writing the minimum generation non-ancient. I've kept around the method `PlatformData.getMinimumGenerationNonAncient()` as a convenience, but that data is no longer directly in the class.